sdk/keyvault/keyvault-common/src/challengeBasedAuthenticationPolicy.ts

```diff
@@ -117,6 +121,23 @@ export function createChallengeCallbacks(): ChallengeCallbacks {
       throw new Error("Missing scope.");
     }
 
+    if (verifyChallengeResource) {
+      let scopeAsUrl: URL;
```

Comment:
I think it would help readability if this logic is extracted to its own helper function. Here's one example where @xirzec's comment about naming this option `disableVerifyChallengeResource` makes sense. That way you can call the function that does the logic `verifyChallengeResource` and the check `disableVerifyChallengeResource`